Assets/LeapMotionModules/Interaction/Examples/Buttons/InteractionSlider.cs

```diff
@@ -1,173 +1,62 @@
-﻿using Leap.Unity.GraphicalRenderer;
-using UnityEngine;
-using UnityEngine.Events;
-
+﻿using UnityEngine;
+using Leap.Unity.Attributes;
 namespace Leap.Unity.UI.Interaction {
-  /** A physics-enabled button. Activation is triggered by physically pushing the button back to its compressed position. 
-   *  Increasing the horizontal and vertical slide limits allows it to act as a 1D or 2D slider
-   */
-  [RequireComponent(typeof(InteractionBehaviour))]
-  public class InteractionSlider : MonoBehaviour {
-    [Tooltip("The minimum and maximum heights the button can exist at.")]
-    public Vector2 MinMaxHeight = new Vector2(0f, 0.02f);
-    [Tooltip("The height that this button rests at; this value is a lerp in between the min and max height.")]
-    public float RestingHeight = 0.5f;
-    [Tooltip("The minimum and maximum horizontal extents that the slider can slide to.")]
-    public Vector2 HorizontalSlideLimits = new Vector2(0f, 0f);
-    [Tooltip("The minimum and maximum vertical extents that the slider can slide to.")]
-    public Vector2 VerticalSlideLimits = new Vector2(0f, 0f);
-
-    //State Events
-    public UnityEvent OnPress = new UnityEvent();
-    public UnityEvent OnUnpress = new UnityEvent();
+  /** A physics-enabled slider. Sliding is triggered by physically pushing the slider to its compressed position. 
+  *  Increasing the horizontal and vertical slide limits allows it to act as either a 1D or 2D slider
+  */
+  public class InteractionSlider : InteractionButton {
+    [Space][Space]
 
-    //Public State variables
-    public bool isDepressed { get; protected set; }
-    public bool depressedThisFrame { get; protected set; }
-    public bool unDepressedThisFrame { get; protected set; }
+    [Tooltip("The minimum and maximum values that the slider reports on the horizontal axis.")]
+    public Vector2 HorizontalValueRange = new Vector2(0f, 1f);
```

Comment:
I think it's going to still be important to distinguish between a 1D slider and a 2D slider